mreg_cli/outputmanager.py

```diff
@@ -0,0 +1,128 @@
+"""This module contains the output management for the mreg_cli package.
+
+This is a singleton class that manages the output for the CLI. It stores the
+output lines and formats them for display. It also manages the filter for the
+command.
+"""
+
+import re
+from typing import Any, List, Tuple
+
+
+class OutputManager:
+    """Manages and formats output for display.
+
+    This is a singleton class to retain context between calls. It is initiated
+    and cleared in the main function of the CLI, all other calls should only
+    add lines to the output via add_line(), add_formatted_line(), or
+    add_formatted_line_with_source(),
+    """
+
+    _instance = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(OutputManager, cls).__new__(cls)
+            cls._instance._lines = []
+        return cls._instance
+
+    def clear(self) -> None:
+        """Clears the object."""
+        self._lines = []
+        self._filter_re = None
+        self._negate = False
+        self._command = None
```

Comment:
Calling `get_filter()` by itself or before `from_command()` will raise an `AttributeError` due to the attribute `command` not being set. Consider setting the attribute in `clear()`:

```suggestion
        self._command = None
        self.command = None
```

Reply:
Indeed! Fixed.